.circleci/check-commit.sh

```diff
@@ -21,29 +21,72 @@ git config submodule.vlsi/hammer-synopsys-plugins.update none
 git submodule update --init
 status=$(git submodule status)
 
+all_names=()
+
 search () {
     for submodule in "${submodules[@]}"
     do
         echo "Running check on submodule $submodule in $dir"
-        hash=$(echo "$status" | grep $submodule | awk '{print$1}' | grep -o "[[:alnum:]]*")
-        echo "Searching for $hash in origin/master of $submodule"
-        git -C $dir/$submodule branch -r --contains "$hash" | grep "origin/master" # needs init'ed submodules
+        hash=$(echo "$status" | grep "$dir.*$submodule " | awk '{print$1}' | grep -o "[[:alnum:]]*")
+        echo "Searching for $hash in origin/$branch of $submodule"
+        (git -C $dir/$submodule branch -r --contains "$hash" | grep "origin/$branch") && true # needs init'ed submodules
```

Comment:
`&& true` is essentially a no-op, and the subshell is unnecessary.  If simply testing for a match, use `grep -q` to suppress output.  Also consider using `grep -F` for fixed-string patterns.

Reply:
Okay, for the record: `&& true` suppresses `-o pipefail`, which seems to follow the standard `-e` behavior about when non-zero exit codes are ignored.

From POSIX:
> When this option is on, if a simple command fails for any of the reasons listed in Consequences of Shell Errors or returns an exit status value >0, and is not part of the compound list following a while, until, or if keyword, and is not a part of an **AND or OR list**, and is not a pipeline preceded by the ! reserved word, then the shell shall immediately exit.